common/beerocks/bcl/include/bcl/network/socket.h

```diff
@@ -43,28 +43,83 @@ typedef unsigned __int64 ssize_t;
 
 class Socket {
 public:
-    explicit Socket(SOCKET s, long readTimeout = 1000);
-    Socket(SOCKET s, const std::string &peer_ip, int port_port, long readTimeout = 1000);
-    Socket(const std::string &uds_path = std::string(), long readTimeout = 1000);
+    // explicit Socket(SOCKET s, long readTimeout = 1000);
+    // Socket(SOCKET s, const std::string &peer_ip, int port_port, long readTimeout = 1000);
+    // Socket(const std::string &uds_path = std::string(), long readTimeout = 1000);
+    static Socket *socketFactory(SOCKET s, long readTimeout = 1000);
     virtual ~Socket();
```

Comment:
Does this work?
```suggestion
    virtual ~Socket() = default;
```
(We only want to declare it virtual, we don't want to provide an implementation.)